src/sinks/loki/sink.rs

```diff
@@ -253,7 +253,15 @@ impl EventEncoder {
         self.remove_label_fields(&mut event);
 
         let timestamp = match event.as_log().get_timestamp() {
-            Some(Value::Timestamp(ts)) => ts.timestamp_nanos_opt().expect("Timestamp out of range"),
+            Some(Value::Timestamp(ts)) => match ts.timestamp_nanos_opt() {
+                Some(timestamp) => timestamp,
+                None => {
+                    let finalizers = event.take_finalizers();
```

Comment:
I think you already have the `finalizers` separated above on line 250.

Reply:
sorry, I missed this line